website_sale_hide_price/views/website_sale_template.xml

```diff
@@ -32,16 +34,21 @@
         <!-- adding website, website.website_show_price and product.website_hide_price to t-cache in order to update cache if parameter modified
              this is needed if you connect with a different user that should not see the prices
              see https://github.com/OCA/e-commerce/issues/818 -->
-        <xpath expr="t[@t-cache='pricelist,product']" position="attributes">
-          <attribute
+        <xpath
+            expr="//t[starts-with(@t-cache, 'pricelist,product')]"
+            position="attributes"
+        >
+            <attribute
                 name="t-cache"
-            >pricelist,product,website,website.website_show_price,product.website_hide_price</attribute>
+                add="website,website.website_show_price,product.website_hide_price"
+                separator=","
+            />
         </xpath>
         <xpath expr="//a[@id='add_to_cart']" position="before">
             <t t-set="user_authenticated" t-value="user_id != website.user_id" />
             <div
                 class="alert alert-info"
-                t-if="product.website_hide_price and (product.website_hide_price_message or website.website_hide_price_default_message) and user_authenticated"
+                t-if="(not website.website_show_price or product.website_hide_price or not user_id.sudo().partner_id.website_show_price) and (product.website_hide_price_message or website.website_hide_price_default_message) and user_authenticated"
```

Comment:
This isn't meant to be triggered by the general setting. The aim is that independently of the general setting we can either show or not custom messages on selected products, but we could hide thos messages without deleting the message itself.

Reply:
Reverted it to as it was before.
Thanks.